src/core_plugins/timelion/public/services/tick_formatters.js

```diff
@@ -18,44 +18,29 @@ function baseTickFormatter(value, axis) {
   return formatted;
 }
 
+const UNIT_PREFIXES = ['', 'k', 'm', 'g', 't', 'p'];
+
+function unitFormatter(unit, divisor, prefixUppercase = false) {
```

Comment:
fwiw, I like the code-reduction, but this meta-programming to dynamically generate this function makes it harder to read. I'd just replace the unit and prefixUppercase arguments with a single label-arg that contains all the labels, all spelled out and explicit.

```
function makeUnitFormatter(divisor, labels) {
 ...
}
```

and then

```...
'bits': makeUnitFormatter(1000, ['b', 'kb', 'mb', 'gb', 'tb', 'pb'])
...
```

personal preference, for your consideration

Reply:
Would agree with you. That solution is way better readable, and even safes some lines of code. So changing it.